framework/src/transfers/MultiAppFieldTransfer.C

```diff
@@ -59,8 +67,8 @@ MultiAppFieldTransfer::transferDofObject(libMesh::DofObject * to_object,
         dof_id_type dof = to_object->dof_number(to_var.sys().number(), to_var.number() + vc, comp);
         dof_id_type from_dof =
             from_object->dof_number(from_var.sys().number(), from_var.number() + vc, comp);
-        Real from_value = from_var.sys().solution()(from_dof);
-        to_var.sys().solution().set(dof, from_value);
+        Real from_value = from_solution(from_dof);
+        to_solution.set(dof, from_value);
```

Comment:
We could transfer from an arbitrary solution to an arbitrary solution. But we have to check if the solution is compatible with the variables. 

I want to understand the use case. What use case you have?

Reply:
If the variables are on the same mesh with same FE family, they are compatible, right? The code does check those. My use case is to transfer main app solution to subapp tagged aux vector and the subapp uses it as the source for a steady-state calculation with `Steady`.